From e54fb13dc4f71fabab5d15fe6db86bcb625c800a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:45 -0400
Subject: [ethernet] igb: fix calls to skb_set_hash

Message-id: <1409062359-13063-28-git-send-email-sassmann@redhat.com>
Patchwork-id: 89148
O-Subject: [RHEL7.1 PATCH 27/81] net: igb calls skb_set_hash
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Tom Herbert <therbert@google.com>

Drivers should call skb_set_hash to set the hash and its type
in an skbuff.

Signed-off-by: Tom Herbert <therbert@google.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 42bdf083fe7017ff0233803175117a54d88eb540)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 3384156..a96beb6 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -6620,7 +6620,9 @@ static inline void igb_rx_hash(struct igb_ring *ring,
           struct sk_buff *skb)
 {
  if (ring->netdev->features & NETIF_F_RXHASH)
-  skb->rxhash = le32_to_cpu(rx_desc->wb.lower.hi_dword.rss);
+  skb_set_hash(skb,
+        le32_to_cpu(rx_desc->wb.lower.hi_dword.rss),
+        PKT_HASH_TYPE_L3);
 }
 
 /**
-- 
1.7.1